File: Dataset/Leetcode/train/62/544.java
class Solution {
    public int XXX(int m, int n) {
        int[][] a = new int[m+1][n+1];
        //第一行第一列只有一条路劲
        for(int i=1;i<=n;i++){
            a[1][i] = 1;
        }
        for(int i=1;i<=m;i++){
            a[i][1] = 1;
        }
        //策略a[x][y] = a[x-1][y]+a[x][y-1]
        //正向递推、同时维护a数组
        for(int i=2;i<=m;i++){
            for(int j=2;j<=n;j++){
                a[i][j] = a[i-1][j]+a[i][j-1];
            }
        }
        return a[m][n];
    }
}

File: Dataset/Leetcode/train/67/544.cpp
class Solution {
public:
    string XXX(string a, string b) {
        int len1=a.length(),len2=b.length();
        if(len2>len1){
            swap(a,b);
            swap(len1,len2);
        }
        int i=len1,j=len2;
        string ans(i+1,'0');
        int jinwei=0;
        while(j>0){
            int add=(a[i-1]-'0')+(b[j-1]-'0')+jinwei;
            ans[i]=add%2+'0';
            jinwei=add>1?1:0;
            i--;
            j--;
        }
        while(i>0){
            int add=a[i-1]-'0'+jinwei;
            ans[i]=add%2+'0';
            jinwei=add>1?1:0;
            i--;
        }
        if(jinwei==1){
            ans[i]='1';
            return ans;
        }
        else return ans.substr(1);
    }
};

